lib/rules/headers/copyright.js

```diff
@@ -37,120 +48,165 @@ const linksToCheckBase = {
+function checkLatestOrLegacyCopyright(sr, copyright, licenseTexts) {
+    const legacyRegex = getLegacyCopyrightMatchRegex(sr, licenseTexts);
+    if (sr.norm(copyright.textContent).match(legacyRegex)) {
+        sr.warning(self, 'legacy-copyright-found', {
+            text: sr.norm(copyright.textContent),
+            rex: legacyRegex,
         });
+        checkLegacyCopyright(sr, copyright, licenseTexts);
+        return;
+    }
 
-        // allowBothLicense: copyright can claim either license.
-        if (licenseTexts.length === 2) allowBothLicense = true;
-        const licenseRex = licenseTexts.join('|');
-
-        const year = (sr.getDocumentDate() || new Date()).getFullYear();
-        const startRex = `^Copyright [©|&copy;] (?:(?:199\\d|20\\d\\d)-)?${year} *W3C[®|&reg;] \\(MIT, ERCIM, Keio, Beihang\\)`;
-        const endRex = `\\. W3C liability, trademark,? and (${licenseRex}) rules apply\\.$`;
-
-        // check text of copyright
-        const rex = new RegExp(startRex + endRex);
-        const text = sr.norm(copyright.textContent);
-        const regResult = text.match(rex);
-        if (!regResult) {
-            sr.error(self, 'no-match', { rex });
-            return done();
-        }
+    checkLatestCopyright(sr, copyright, licenseTexts);
+}
 
-        const [, licenseInDoc] = regResult;
+/**
+ * @param sr
+ * @param done
+ */
+export async function check(sr, done) {
+    const copyright = sr.jsDocument.querySelector('body div.head p.copyright');
 
-        // When document can have 2 licenses, remove one of the link which is not mentioned in document.
-        if (allowBothLicense) {
-            if (licenseInDoc === LICENSE_CS_TEXT) {
-                delete linksToCheck[LICENSE_CD_TEXT];
-            } else if (licenseInDoc === LICENSE_CD_TEXT) {
-                delete linksToCheck[LICENSE_CS_TEXT];
-            }
-        }
+    if (!copyright) {
+        sr.error(self, 'not-found');
+        return done();
+    }
+
+    if (await isOnlyPublishedByTagOrAb(sr)) {
+        return done();
+    }
+
+    const chartersData = await sr.getChartersData();
+    if (!chartersData || !chartersData.length) {
+        sr.error(self, 'no-data-from-API');
+        return done();
+    }
+
+    const allowedLicenses = getCommonLicenseUri(chartersData);
+    if (!allowedLicenses.length && chartersData.length > 1) {
+        sr.error(self, 'no-license-found-joint');
+        return done();
+    }
+    if (!allowedLicenses.length) {
+        sr.error(self, 'no-license-found');
+        return done();
+    }
+
+    // licenseTexts: ['permissive document license'] or ['document use'] or ['document use', 'permissive document license']
+    const licenseTexts = allowedLicenses
+        .filter(v => [LICENSE_CD_URL, LICENSE_CS_URL].includes(v))
+        .map(v => LICENSE_URL_TEXT_MAP[v]);
+
+    sr.transition({
+        from: new Date(2023, 0, 1),
+        to: new Date(2023, 1, 1),
```

Comment:
The dates comparison in `transition` do not include the limits
```suggestion
        to: new Date(2023, 0, 31),
```